### PR TITLE
User Management - Add Invitation apis

### PR DESCRIPTION
**Pull request by jbvolta**
## Description
User Management - Add Invitation apis

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: lib/UserManagement.php [hunk 1]
```diff
@@ -257,6 +257,9 @@ public function listOrganizationMemberships(
         $params = [
             "organization_id" => $organizationId,
             "user_id" => $userId,
+            "limit" => $limit,
```
Comment (jbvolta) [inline lib/UserManagement.php:260]: not related but I realized I missed those in #190.
I can extract to another PR if you prefer.

Comment (amygdalama) [reply lib/UserManagement.php:260]: no worries, good catch

File: lib/UserManagement.php [hunk 2]
```diff
@@ -278,6 +281,134 @@ public function listOrganizationMemberships(
...
+     *
+     * @param string $email The email address of the invitee
+     * @param string $organizationId Organization ID
+     * @param int $expiresInDays expiration delay in days
+     * @param string $inviterUserId User ID of the inviter
+     * @throws Exception\WorkOSException
+     *
+     * @return \WorkOS\Resource\Invitation
+     */
+    public function sendInvitation($email, $organizationId, $expiresInDays, $inviterUserId)
```
Comment (amygdalama) [inline lib/UserManagement.php:295]: `organizationId`, `expiresInDays`, and `inviterUserId` should be optional

File: lib/UserManagement.php [hunk 3]
```diff
@@ -278,6 +281,134 @@ public function listOrganizationMemberships(
...
+            true
+        );
+
+        return Resource\Invitation::constructFromResponse($response);
+    }
+
+    /**
+     * List Invitations
+     *
+     * @param string $invitationId ID of the Invitation
```
Comment (amygdalama) [inline lib/UserManagement.php:343]: copy / pasta

File: lib/UserManagement.php (hunk 1 above)
Comment (amygdalama) [reply lib/UserManagement.php:260]: oh i just noticed the return type in the comment of this method is inaccurate; can be fixed here or in a followup, whatever you prefer

File: lib/UserManagement.php [hunk 4]
```diff
@@ -278,6 +281,134 @@ public function listOrganizationMemberships(
...
+    /**
+     * Sends an Invitation
+     *
+     * @param string $email The email address of the invitee
+     * @param string $organizationId Organization ID
+     * @param int $expiresInDays expiration delay in days
+     * @param string $inviterUserId User ID of the inviter
+     * @throws Exception\WorkOSException
+     *
+     * @return \WorkOS\Resource\Invitation
```
Comment (amygdalama) [inline lib/UserManagement.php:293]: i noticed for some other endpoints we use
```
@return \WorkOS\Resource\Response
```

is one better or more accurate than the other?

File: lib/UserManagement.php (hunk 1 above)
Comment (jbvolta) [reply lib/UserManagement.php:260]: good catch too 😄

Comment (amygdalama) [reply lib/UserManagement.php:260]: 🙌

File: lib/UserManagement.php (hunk 4 above)
Comment (jbvolta) [reply lib/UserManagement.php:293]: Response is before parsing, it's the raw http request. I want to return the parsed entities, which are all extending BaseWorkOSResource

Comment (amygdalama) [reply lib/UserManagement.php:293]: got it. so i think that means the other methods denoted with `@return \WorkOS\Resource\Response` are inaccurate? can you review them in a followup please 🙏

Comment (jbvolta) [reply lib/UserManagement.php:298]: I'll check the UserManagement.

Comment (amygdalama) [reply lib/UserManagement.php:298]: perfect thanks!

File: lib/UserManagement.php [hunk 5]
```diff
@@ -278,6 +285,147 @@ public function listOrganizationMemberships(
         return [$before, $after, $organizationMemberships];
     }
 
+    /**
+     * Sends an Invitation
+     *
+     * @param string $email The email address of the invitee
+     * @param string $organizationId|null Organization ID
+     * @param int $expiresInDays|null expiration delay in days
+     * @param string $inviterUserId|null User ID of the inviter
```
Comment (amygdalama) [inline lib/UserManagement.php:294]:
```suggestion
     * @param string|null $organizationId Organization ID
     * @param int|null $expiresInDays expiration delay in days
     * @param string|null $inviterUserId User ID of the inviter
```

**Review by amygdalama — APPROVED** [review]
one minor suggestion otherwise looks good!

**Pull request merged**